### PR TITLE
Add private endpoint for eventgrid

### PR DESCRIPTION
**Pull request by janboll**
### What this PR does

Jira: 
Link to demo recording:

### CODE REVIEW

File: dev-infrastructure/modules/private-endpoint.bicep [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      subnet: {
+        id: aksNodeSubnetId
+      }
+    }
+  }
+]
+
+resource privateEndpointDnsGroup 'Microsoft.Network/privateEndpoints/privateDnsZoneGroups@2023-09-01' = [
+  for index in range(0, length(subnetIds)): {
+    name: '${serviceType}-dns-group'
```
Comment (geoberle) [inline dev-infrastructure/modules/private-endpoint.bicep:41]: would this name to be made unique by adding uniqueString(subnetIds) to it?

File: dev-infrastructure/modules/maestro/maestro-infra.bicep [hunk 2]
```diff
@@ -247,5 +254,14 @@ resource maestroConsumersPublishTopicspacePermissionBinding 'Microsoft.EventGrid
...
 
+var privateDnsZoneName = 'privatelink.ts.eventgrid.azure.net'
+
+// TODO: find a place for this
+resource eventGridPrivateEndpointDnsZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
+  name: privateDnsZoneName
+  location: 'global'
+  properties: {}
+}
+
```
Comment (geoberle) [inline dev-infrastructure/modules/maestro/maestro-infra.bicep:265]: you should be able to move this into `private-endpoint.bicep` as well

File: dev-infrastructure/Makefile [hunk 3]
```diff
@@ -199,6 +199,10 @@ svc.wait:
 
 svc: svc.wait svc.rg
 	@scripts/cleanup-orphaned-rolebindings.sh $(SVC_RESOURCEGROUP)
+cluster: 
+ifndef AKSCONFIG
+	$(error "Must set AKSCONFIG")
+endif
```
Comment (geoberle) [inline dev-infrastructure/Makefile]: not required anymore since #762 is now merged

File: dev-infrastructure/modules/private-endpoint.bicep [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+param privateLinkServiceId string
+
+param groupIds array
+
+param privateEndpointDnsZoneName string
+
+param vnetId string
+
+resource eventGridPrivateEndpointDnsZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
```
Comment (geoberle) [inline dev-infrastructure/modules/private-endpoint.bicep:14]: remove `eventGrid` prefixes from resources in this module. this is reusable for general private endpoint stuff, right?

File: dev-infrastructure/templates/region.bicep [hunk 5]
```diff
@@ -66,5 +66,6 @@ module maestroInfra '../modules/maestro/maestro-infra.bicep' = {
     maxClientSessionsPerAuthName: maestroEventGridMaxClientSessionsPerAuthName
     maestroKeyVaultName: maestroKeyVaultName
     kvCertOfficerManagedIdentityName: maestroKeyVaultCertOfficerMSIName
+    publicNetworkAccess: 'Disabled'
```
Comment (geoberle) [inline dev-infrastructure/templates/region.bicep:69]: let's keep eventgrid public for now while also using the private endpoint

Comment (janboll) [reply dev-infrastructure/templates/region.bicep:69]: Any particular reason?

**Review by geoberle — APPROVED** [review]

**Pull request merged**